scipy/sparse/_bsr.py

```diff
@@ -485,7 +485,7 @@ def tocoo(self, copy=True):
                 raise ValueError("Matrix too big to convert")
             indptr_diff = indptr_diff_limited
 
-        row = (R * np.arange(M//R)).repeat(indptr_diff)
+        row = (R * np.arange(M//R)).repeat(indptr_diff)  # Should this be a smaller dtype?
```

Comment:
Does this produce int64 indices unconditionally?

Reply:
I believe so, but I'm not familiar enough with the BSR format well enough to be 100% sure. 

Is there any way `bsr.shape` or `bsr.blocksize` would be `floating`?

Reply:
There were a few tests that were receiving floating values. I'm 90% sure those were only coming from `_min_or_max`